### DISCUSSION TITLE
[Bug] Scoop should never allow wipping the whole environment variables....

### DISCUSSION THREAD

**Discussion by melMass** [ScoopInstaller/Scoop#5666] [Help and Questions] [answered]
#### Current Behavior
I will add more info later on, currently everything broke and imo this should not happen without a clear warning.

#### Expected Behavior
If a formula doesn't explicitly extend path but overwrite it it should at the very least warn the user

```
"env_set": {
        "PATH": "$dir\\bin"
    }
```

#### Possible Solution
Warn, fail, disallow completely (unless I'm missing something this is prone to errrors).

### System details

**Windows version:** [e.g. 7, 8, 10, 11]

**OS architecture:** [e.g. 32bit, 64bit, arm64]

**PowerShell version:** [output of `"$($PSVersionTable.PSVersion)"`]

**Additional software:** [(optional) e.g. ConEmu, Git]

#### Scoop Configuration

```json
//# Your configuration here
```

**Comment by rashil2000** [comment, marked as answer]
Scoop manifests are just config files for a PowerShell scripts - they obviously have the potential to carry out very destructive actions. Overwriting PATH is one of them. A manifest that is directly messing with the PATH (instead of using fields like `env_add_path`) should not even be allowed in a bucket.

Unfortunately, it is not possible to have checks for such a wide variety of destructive actions.

**Reply by melMass** [reply]
Thanks it makes sense that manifest are double checked for these kind of things before being included in a bucket.
This is definitely user error btw (from my side) as I was writing a manifest and though it was strange that setting env_set.PATH would completely replace it (I'm still a windows noob and I really don't like the registry based env variables concept exactly because of cases like this, I also had that exact issue installing Fing desktop)

**Reply by melMass** [reply]
I guess my initial issue was about having some kind of protection against key variables when using `env_set`, but I get your perspective about the flexibility/responsibility aspects
